ast/src/call.rs

```diff
@@ -43,6 +46,25 @@ pub struct ProgrammaticCall<'a> {
     pub type_parameters: Vec<Type<'a>>,
 }
 
+/// A method call.
+#[segment_holder]
+#[derive(Debug, Clone, PartialEq, DebugPls)]
+pub struct MethodCall<'a> {
+    /// The expression on which the method is called.
+    pub source: Box<Expr<'a>>,
+
+    /// The name of the method to call.
+    ///
+    /// The name cannot be an expression, so it is always a constant string after the parsing phase.
+    pub name: &'a str,
```

Comment:
Use an optional here and do not represent lambda calls (ex: `$a()`) by `ProgrammaticCall`, but use a `MethodCall` with no name instead.